packages/databyss-ui/primitives/Control/native/RichTextInput.js

```diff
@@ -1,11 +1,34 @@
-import React, { forwardRef } from 'react'
+import React, { forwardRef, useEffect, useRef } from 'react'
+import { ReactEditor } from 'slate-react'
+import { Editor, Transforms } from '@databyss-org/slate'
 import SingleLine from '@databyss-org/editor/components/SingleLine'
 
 const RichTextInput = forwardRef(
-  ({ value, onChange, id, concatCss, onBlur, multiline }, ref) => {
+  (
+    { value, onChange, id, concatCss, onBlur, onFocus, active, multiline },
+    ref
+  ) => {
+    const editorRef = useRef(null)
+
     const _onBlur = event => {
       onBlur(event)
-      //  setTimeout(() => editor.blur(), 50)
+    }
+
+    useEffect(
```

Comment:
We should keep slate-specific code in `SingleLine`. How about we pass `active` into `SingleLine` instead of `setEditor` and move this `useEffect` inside?